### PR TITLE
Automatically join channels

### LINKED ISSUE #18: Automatically join channels on start
The client should join pre-defined channels at startup, after Orbit is connected successfully.

Where to define the channels is to be decided (can be a json file, in ipfs, in orbit-db, etc.)

### PR DESCRIPTION
**Pull request by shamb0t**
Joins list of channels from localstorage on start. If none exist, joins '#ipfs'.
closes #18

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -160,6 +163,20 @@ var App = React.createClass({
     document.title = `#${channel}`;
     logger.debug("Set title: " + document.title);
     AppActions.setCurrentChannel(channel);
+
+    let channels = JSON.parse(localStorage.getItem( "anonet.app." + this.state.user.username + "." + this.state.networkName + ".channels")) || [];
```
Comment (dignifiedquire) [inline client/src/components/App.js]: The string concatenation could be extracted into a function like `makeChannelName(username, networkName)` as it's used quite often below

**Comment by shamb0t** [comment]
Nice much cleaner now

File: client/src/components/App.js [hunk 2]
```diff
@@ -118,8 +119,13 @@ var App = React.createClass({
       AppActions.setLocation("Connect");
     } else {
       this.setState({ networkName: network.name });
+      const channels = JSON.parse(localStorage.getItem( "anonet.app." + network.user.username + "." + network.name + ".channels")) || [{ 'name': 'ipfs'}];
+      channels.map( (c) => NetworkActions.joinChannel(c.name, ''));
```
Comment (haadcode) [inline client/src/components/App.js]: Use `.forEach` here as you're looping through the channels but not returning anything.

**Comment by haadcode** [comment]
This is great @shamb0t! While we should figure out a good place for the logic, this looks good to me. Will test it later tonight and merge to `orbitdbrefactor` afterwards.

On that note, @shamb0t, I can't merge this and the emoji preview to master yet due to working-on-everything-and-the-whole-world-at-once, so things are in flux atm a bit. Hopefully will be able to put it all together soon (next week or so).

**Comment by haadcode** [comment]
I already merged this before trying. Made a small change to your implementation: don't join #ipfs as default. It's a little confusing to be dropped into the chat immediately and I think a better first time experience is through the action of joining a channel (manually).

**Pull request merged**